compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -365,39 +419,75 @@ abstract class PrepJSInterop extends plugins.PluginComponent
           val tSym = sym.tpe.typeSymbol
 
           tSym.setAnnotations(rawJSAnnot :: sym.annotations)
-
       }
 
-      if (shouldPrepareExports) {
-        // Check that RawJS type is not exported
+      if (isJSNative && shouldPrepareExports) {
+        // Check that a JS native type is not exported
         for {
           exp <- exportsOf(sym)
           if !exp.ignoreInvalid
-        } reporter.error(exp.pos, "You may not export a class extending js.Any")
+        } {
+          reporter.error(exp.pos,
+              "You may not export a native JS class or object")
+        }
       }
 
       if (shouldCheckLiterals)
         checkJSNameLiteral(sym)
 
-      if (implDef.isInstanceOf[ModuleDef])
-        enterJSAnyMod { super.transform(implDef) }
-      else
-        enterJSAnyCls { super.transform(implDef) }
+      val kind = {
+        if (!isJSNative) OwnerKind.JSClass
+        else if (implDef.isInstanceOf[ModuleDef]) OwnerKind.JSNativeMod
+        else OwnerKind.JSNativeClass
+      }
```

Comment:
From this I read that there is no such thing as a Scala.js defined JS module (which is reasonable IMO). However, I have not found the guard against that in this phase.